external_executors/local_cromwell/local_cromwell.py

```diff
@@ -0,0 +1,255 @@
+table = ddb.Table("{db_table}")
+table.put_item(
+    Item={{
+        "EventId": event_id,
+        "JobId": "{job_id}",
+        "EventType": "$1",
+        "Message": "$2" or "empty",
+        "Time": "{{:%Y-%m-%dT%H:%M:%SZ}}".format(datetime.datetime.utcnow())
+    }}
+)
+END
+}}
+
+function fail {{
+    set +e
+    record_event "JobFailed"
+    shutdown -H 5
+}}
+
+trap "fail" EXIT
+
+# Get dependencies for local cromwell
+yum update -y
+yum install -y docker java-1.8.0-openjdk-headless python36 python36-pip
+pip-3.6 install boto3
+service docker start
+wget -q https://github.com/broadinstitute/cromwell/releases/download/30.1/cromwell-30.1.jar
+
+# Record a job start event
+record_event "JobStarted"
+
+# The inputs arrive as a JSON string, but write them to a file
+echo {inputs_json} > inputs.json
+
+# Similarly, the WDL comes as a string, so write it to a file Cromwell will read
+echo {wdl} > workflow.wdl
+
+# Finally, decode the base64-encoded dependencies into a file and unzip them.
+if [ ! -z "{dependencies}" ]; then
+    echo "{dependencies}" | base64 -d - > dependencies.zip
+    unzip dependencies.zip
+fi
+
+# Iterate over the inputs and download any files. Then replace the entries
+# that refer to remote locations with entries that refer to the local paths.
+python3 << END
+import json, os, subprocess
+inputs_dict = json.load(open("inputs.json"))
+localized_inputs_dict = dict()
+for entry, value in inputs_dict.items():
+    if value.startswith("https://") or value.startswith("http://"):
+        local_path = os.path.basename(value)
+        subprocess.run(["curl", "-o", local_path, value], check=True)
+        localized_inputs_dict[entry] = local_path
+    else:
+        localized_inputs_dict[entry] = value
+json.dump(localized_inputs_dict, open("localized_inputs.json", "w"))
+END
+
+# Record an inputs localized event
+record_event "InputsLocalized"
+
+# Run Cromwell
+record_event "CromwellStarting"
+set +e
+java8 -jar cromwell-30.1.jar run workflow.wdl --inputs localized_inputs.json 2>&1
+cromwell_rc=$?
+set -e
+record_event "CromwellFinished" "$cromwell_rc"
+
+# Record a log event
+python3 << END
+import boto3, datetime, glob, json, uuid
+event_id = str(uuid.uuid4())
+ddb = boto3.resource("dynamodb", region_name="{region}")
+table = ddb.Table("{db_table}")
+try:
+    log_path = glob.glob("cromwell-workflow-logs/*log")[0]
+    log = json.dumps(open(log_path).read())
+except IndexError:
+    log = "No Cromwell logs found."
+    
+table.put_item(
+    Item={{
+        "EventId": event_id,
+        "JobId": "{job_id}",
+        "EventType": "JobLog",
+        "Message": log,
+        "Time": "{{:%Y-%m-%dT%H:%M:%SZ}}".format(datetime.datetime.utcnow())
+    }}
+)
+END
+
+if [ $cromwell_rc == 0 ]; then
+    record_event "JobSucceeded"
+else
+    record_event "JobFailed"
+fi
+
+trap - EXIT
+shutdown -H 5
+"""
+
+# Define boto3 clients
+EC2 = boto3.client('ec2', region_name=REGION)
+
+def workflows_post(event, context):
```

Comment:
Might be helpful to note the python type of event (dict?) and context in the docstring.

Reply:
Okay, I'll expand the docstring a little.